snuba_sdk/join.py

```diff
@@ -0,0 +1,54 @@
+from dataclasses import dataclass
+from typing import Any, MutableMapping, Sequence, Tuple
+
+from snuba_sdk.entity import Entity
+from snuba_sdk.expressions import Expression, InvalidExpression
+
+
+@dataclass(frozen=True)
+class Relationship(Expression):
+    lhs: Entity
+    name: str
+    rhs: Entity
+
+    def validate(self) -> None:
+        def valid_entity(e: Any) -> None:
+            if not isinstance(e, Entity):
+                raise InvalidExpression(f"'{e}' must be an Entity")
+            elif e.alias is None:
+                raise InvalidExpression(f"{e} must have a valid alias")
+
+        valid_entity(self.lhs)
+        valid_entity(self.rhs)
+
+        if not isinstance(self.name, str) or not self.name:
+            raise InvalidExpression(f"'{self.name}' is not a valid relationship name")
+
+
+@dataclass(frozen=True)
+class Join(Expression):
+    relationships: Sequence[Relationship]
+
+    def get_alias_mappings(self) -> Sequence[Tuple[str, str]]:
```

Comment:
Hmm this might have been a holdover. This was originally a Sequence[str] so that I could more easily track duplicates but now I think I don't need that.